libvncclient/sockets.c

```diff
@@ -390,6 +456,17 @@ ConnectClientToTcpAddr(unsigned int host, int port)
 
 rfbSocket
 ConnectClientToTcpAddr6(const char *hostname, int port)
+{
+  rfbSocket sock = ConnectClientToTcpAddr6WithTimeout(hostname, port, SOCKET_DEFAULT_TIMEOUT);
+  /* put socket back into blocking mode for compatibility reasons */
+  if (sock != RFB_INVALID_SOCKET) {
+    SetBlocking(sock);
```

Comment:
what if the socket was nonblocking before?

Reply:
If I'm not mistaken the `ConnectClient*()` functions until now always returned sockets in blocking mode so I tried to imitate this behaviour for compatibility reasons. Only the more highlevel `ConnectToRFBServer()` and `ConnectToRFBRepeater()` change the socket to non-blocking mode.